### PR TITLE
fix: update references from twitter.com to x.com

### LINKED ISSUE #2198: Update Reference Links from twitter to x
### Preliminary Checks

- [X] I have reviewed the documentation: https://unkey.com/docs

- [X] I have searched for existing issues: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Reproduction / Replay Link (Optional)

_No response_

### Issue Summary

With the rebranding of Twitter to X, all references in our codebase that point to `twitter.com` need to be updated to `x.com` to ensure accuracy and consistency.

### Steps to Reproduce

1. Clone the repository or navigate to the existing local copy.
2. Open the codebase in your preferred text editor or IDE.
3. Search for instances of twitter.com in the codebase.
4. Note the locations of all instances where links or references are made to twitter.com.

### Expected behavior

1. All instances of twitter.com should be updated to x.com.
2. The application should function as expected without any broken links or references.

### Other information

_No response_

### Screenshots

_No response_

### Version info

```markdown
- OS: .
- Node: .
- npm: .
```

### PR DESCRIPTION
**Pull request by Vardhaman619**
## What does this PR do?

This PR updates all references from `twitter.com` to `x.com` in the codebase.

Fixes #2198 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [X] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

To verify the changes, please perform the following steps:
1. Clone the repository and checkout to this branch.
2. Search the codebase for any remaining instances of `twitter.com` to ensure all have been updated to `x.com`.
3. Run the application to verify that all links function correctly without errors.

### Required

- [X] Filled out the "How to test" section in this PR
- [X] Read [Contributing Guide](./CONTRIBUTING.md)
- [X] Self-reviewed my own code
- [X] Commented on my code in hard-to-understand areas
- [X] Ran `pnpm build`
- [X] Ran `pnpm fmt`
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] Merged the latest changes from main onto my branch with `git pull origin main`
- [X] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Release Notes

- **New Features**
	- Introduced usage-based invalidation for API keys, allowing users to limit requests.
	- Added instructions for updating API keys with new rate limits and metadata.

- **Updates**
	- Updated references from Twitter to X.com across various components and documentation.
	- Minor grammatical and stylistic enhancements in documentation.

- **Bug Fixes**
	- Corrected URLs in quotes and social media links to reflect platform rebranding.

### CODE REVIEW

**Review by chronark — APPROVED** [review]

**Comment by chronark** [comment]
Please resolve the merge conflict, /apps/engineering/mint.json should be deleted

**Comment by chronark** [comment]
why do you delete files in apps/engineering?

**Comment by Vardhaman619** [comment]
@chronark I attempted to resolve the merge conflicts, but I seem to have made some unintended changes, particularly regarding the deletion of files in the /apps/engineering/ directory.

Could you please guide me on what steps I missed or clarify which files exactly should remain or be deleted? I will make the necessary corrections promptly.

File: apps/www/content/blog/how-unkey-treats-marketing.mdx [hunk 1]
```diff
@@ -166,7 +166,7 @@ Before you get upset, the community needs these events. They are the lifeblood o
 
 - Frontend Horse Christmas Spectacular
 - Github projects
-- Twitter Guesser game show... Which was hilarious.
+- X(Twitter) Guesser game show... Which was hilarious.
```
Comment (perkinsjr) [inline apps/www/content/blog/how-unkey-treats-marketing.mdx:169]: This shouldn't be changed.

**Review by perkinsjr — CHANGES_REQUESTED** [review]
See my comment

File: apps/www/content/blog/how-unkey-treats-marketing.mdx (hunk 1 above)
Comment (Vardhaman619) [reply apps/www/content/blog/how-unkey-treats-marketing.mdx]: ok

**Comment by Vardhaman619** [comment]
@perkinsjr Could you please take a look at this PR

**Comment by chronark** [comment]
there are merge conflicts

**Comment by Vardhaman619** [comment]
@chronark / @perkinsjr Could you please take a look at this PR

**Review by perkinsjr — APPROVED** [review]

**Pull request merged**